### PR TITLE
Duplo 15257 Elastic cache support

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Added support attribute for setting cluster mode and number of shards
## Summary of changes
Added support attribute NumberOfShards and EnableClusterMode
This PR does the following:

- Added schema attribute enable_cluster_mode and number_of_shards
- Added request/response attribute EnableClusterMode and NumberOfShards
- Updated documents

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ✓] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-15257 TF: Add support for `Enable cluster mode` and `no. of shards` for ecache_instance resource](https://app.clickup.com/t/8655600/DUPLO-15257)

File: duplocloud/resource_duplo_ecache_instance.go [hunk 1]
```diff
@@ -298,6 +312,13 @@ func expandEcacheInstance(d *schema.ResourceData) *duplosdk.DuploEcacheInstance
 		KMSKeyID:            d.Get("kms_key_id").(string),
 		ParameterGroupName:  d.Get("parameter_group_name").(string),
 	}
+	if data.CacheType == 0 {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_ecache_instance.go:315]: It is better to check if the attribute is part of resource data while reading, something like 
`v, ok := d.GetOk("enable_cluster_mode"); ok`

File: examples/resources/duplocloud_ecache_instance/resource.tf [hunk 2]
```diff
@@ -4,9 +4,11 @@ resource "duplocloud_tenant" "myapp" {
...
-  cache_type = 0 // Redis
-  replicas   = 1
-  size       = "cache.t2.small"
+  tenant_id           = duplocloud_tenant.myapp.tenant_id
+  name                = "mycache"
+  cache_type          = 0 // Redis
+  replicas            = 1
+  size                = "cache.t2.small"
+  enable_cluster_mode = true // applicable only for redis
+  number_of_shards    = 1    // applicable only for redis
```
Comment (tahir-duplo) [inline examples/resources/duplocloud_ecache_instance/resource.tf:13]: Whenever you add new attributes to the existing resource it's better to create a ClickUp ticket to support the same attributes in the exporter.

Comment (Duplo-nikv) [reply examples/resources/duplocloud_ecache_instance/resource.tf:13]: ok

File: duplocloud/resource_duplo_ecache_instance.go (hunk 1 above)
Comment (Duplo-nikv) [reply duplocloud/resource_duplo_ecache_instance.go:315]: ok

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**